aspnetcore/mvc/controllers/routing.md

```diff
@@ -70,667 +65,788 @@ public class HomeController : Controller
+This mapping:
 
-> [!TIP]
-> Using conventional routing with the default route allows you to build the application quickly without having to come up with a new URL pattern for each action you define. For an application with CRUD style actions, having consistency for the URLs across your controllers can help simplify your code and make your UI more predictable.
+* Is based on the controller and [action](#action) names **only**.
+* Isn't based on namespaces, source file locations, or method parameters.
+* Is different than typical [attribute routing](#ar) which doesn't include the action name in the route.
+
+Using conventional routing with the default route allows creating the app without having to come up with a new URL pattern for each action. For an app with [CRUD](https://wikipedia.org/wiki/Create,_read,_update_and_delete) style actions, having consistency for the URLs across controllers:
+
+* Helps simplify the code.
+* Makes the UI more predictable.
 
 Most apps should choose a basic and descriptive routing scheme so that URLs are readable and meaningful. The default conventional route `{controller=Home}/{action=Index}/{id?}`:
+
 * Supports a basic and descriptive routing scheme.
 * Is a useful starting point for UI-based apps.
+* Is the only route template needed for many web UI apps. For larger web UI apps, another route using [Areas](#areas) if frequently all that's needed.
+
+<xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllerRoute*> and <xref:Microsoft.AspNetCore.Builder.MvcAreaRouteBuilderExtensions.MapAreaRoute*> :
+
+* Automatically assign an **order** value to their endpoints based on the order they are invoked.
+* Don't expose the [RouteEndpoint.Order](xref:Microsoft.AspNetCore.Routing.RouteEndpoint.Order*). `RouteEndpoint.Order` is only exposed for [attribute routing](#ar). <!-- review required for preceding two bullets. Are they OK? -->
+
+Endpoint routing in ASP.NET Core 3.0 and later:
+
+* Doesn't have a concept of routes.
+* Doesn't provide ordering guarantees, all endpoints are processed at once.
 
 > [!WARNING]
-> The `id` is defined as optional by the route template, meaning that your actions can execute without the ID provided as part of the URL. Usually what will happen if `id` is omitted from the URL is that it will be set to `0` by model binding, and as a result no entity will be found in the database matching `id == 0`. Attribute routing can give you fine-grained control to make the ID required for some actions and not for others. By convention the documentation will include optional parameters like `id` when they're likely to appear in correct usage.
+> The `id` is defined as optional by the preceding route template. Actions can execute without the optional ID provided as part of the URL. Generally, when`id` is omitted from the URL:
```

Comment:
This warning needs to appear closer to the example, it's totally out of context here